lib/eslint/flat-eslint.js

```diff
@@ -362,17 +362,6 @@ async function calculateConfigArray(eslint, {
             const negated = pattern.startsWith("!");
             const basePattern = negated ? pattern.slice(1) : pattern;
 
-            /*
-             * Ignore patterns are considered relative to a directory
-             * when the pattern contains a slash in a position other
-             * than the last character. If that's the case, we need to
-             * add the relative ignore path to the current pattern to
-             * get the correct behavior. Otherwise, no change is needed.
-             */
-            if (!basePattern.includes("/") || basePattern.endsWith("/")) {
-                return pattern;
-            }
-
```

Comment:
I'm not sure why we had this special case. It could be a leftover from the version where patterns had .gitignore semantics (e.g, `*.js` matches at any level).

Reply:
Yup, I’m pretty sure that’s left over from gitignore compatibility.